### ISSUE TITLE
docs: docker swarm service unique service name

### ISSUE THREAD

**Issue by sonix07** [cockroachdb/cockroach#17133]
I've just read through your very comprehensible docker swarm tutorial (https://www.cockroachlabs.com/docs/stable/orchestrate-cockroachdb-with-docker-swarm-insecure.html#see-also)
I am wondering why every instance needs a unique service name. This makes it hard to leverage many docker swarm features like `--mode global` or `--replicas 5` or `tasks.cockroachdb` (for service discovery) and `docker service update --replicas 9 cockroachdb`.

I am trying to get something like this to work:

```
docker service create \
	--name cockroachdb \
	--network roachnet \
	--replicas 5 \
	--stop-grace-period 60s \
	--publish 8080:8080 \
	cockroachdb/cockroach:v1.0.3 start \
	--join=cockroachdb:26257 `\`
	--logtostderr \
	--insecure
```
My ultimate goal is having a swarm cluster with local storage only (no NFS, NAS or SAN) and leveraging the fault tolerance of the database itself.
The application I want to use with cockroachdb uses a connection string where only one entry can be made. Thanks to docker swarm I can put something like `tasks.servicename` and it will forward the request to a different node (in case of failure or for load balancing) transparent to the app. This gives me great fault tolerance, but the service name has to be the same for each instance/replica.

**In a nutshell**: How can I have a cockroachdb cluster with multiple instances using the same service name (`docker service create --name cockroachdb`).

**Comment by a-robinson** [comment]
There was recently [a similar discussion on our forum](https://forum.cockroachlabs.com/t/docker-swarm-scaling/788/2). Mind reading it and following up with any questions you have afterwards?

The tl;dr is that you should be able to do so today, but it'll be tough to do in secure clusters. You'll just have to do a little extra work when starting the cluster to make sure that it gets initialized.

**Comment by sonix07** [comment]
Unfortunately I could not find out how to use the `cockroach init` instead of `join`.
But I found a workaround for docker swarm by creating two services and removing one after that.

This one I use to bootstrap the cluster. Note that this service is called **cockroachdb0**
```
docker service create \
	--name cockroachdb0 \
	--network roachnet \
	--replicas 1 \
	--stop-grace-period 60s \
	cockroachdb/cockroach:v1.0.3 start \
	--logtostderr \
	--insecure
```
My actual cluster will be this service called **cockroachdb**, also I use the join command to join cockroachdb0 and cockroachdb
```
docker service create \
	--name cockroachdb \
	--network roachnet \
	--replicas 3 \
	--stop-grace-period 60s \
	--publish 8080:8080 \
	cockroachdb/cockroach:v1.0.3 start \
	--join=cockroachdb0:26257 \
	--join=cockroachdb:26257 \
	--logtostderr \
	--insecure
```
After this I simply remove cockroachdb0
```
docker service rm cockroachdb0
```
Now I am able to use the scale (`docker service update --replicas 10 cockroachdb`) command to create more instances as I go and they get added to my cockroachdb cluster. Another very important aspect is that now I can use `tasks.cockroachdb` (perform nslookup tasks.cockroachdb [works only inside roachnet] and I get all available instances) as a connection string for my application and get the full benefit and fault tolerance of swarm and cockroachdb.


This still seems not very elegant to me... In my opinion it would be great if one could just do this with a single command looking similar to the example below. Where 'join' points to the same service name. I've tried this multiple times but I only get a 404 page not found error.

```
docker service create \
	--name cockroachdb \
	--network roachnet \
	--replicas 5 \
	--stop-grace-period 60s \
	--publish 8080:8080 \
	cockroachdb/cockroach:v1.0.3 start \
	--join=cockroachdb:26257 `\`
	--logtostderr \
	--insecure
```

**Conclusion**: It would be very nice if the `join` command could handle being pointed to an instance of itself.

**Comment by a-robinson** [comment]
Yes, the `init` command is going to be in release 1.1, so it's not documented for the 1.0 release. What you did is basically equivalent to what I would have done at this point. I agree with you that the DNS entry you get from having all the nodes be part of the same service is very useful. We'll switch over to this sort of model once 1.1 has been released in a couple months so that we can use the `init` approach.

**Comment by sonix07** [comment]
Would you say that this is a recommended configuration at this point in time? With the exception of the `--insecure` option, but I will use an nginx proxy to authenticate and all the traffic between the nodes is already encrypted due to the docker swarm overlay network

**Comment by a-robinson** [comment]
If you're explicitly ok with the caveats that go along with running an `--insecure` cluster, which are [listed in our docs](https://www.cockroachlabs.com/docs/stable/manual-deployment-insecure.html#recommendations), then it's a recommended configuration.

**Comment by a-robinson** [comment]
Actually, coming back to this, what are you doing for volumes in your new configuration? Writing cockroach's data into each container's filesystem is a very bad practice.

**Comment by sonix07** [comment]
@a-robinson this was just to quickly evaluate the software. Now I am mounting local ssd storage from each node and leveraging the fault tolerance of the application. Do you think that this is a good idea? I'm trying to avoid purchasing a SAN

**Comment by a-robinson** [comment]
Yes, that's reasonable as long as the data is being written to the mounted SSD.

**Comment by a-robinson** [comment]
Preemptive note to anyone looking over 1.1 milestone issues -- this isn't a code change needed for the 1.1 release, it's an orchestration/documentation change to be made shortly after the 1.1 release is available. If that means we should remove the milestone go for it, but in the meantime I find it handy to keep the milestone on it.

**Comment by knz** [comment]
Is there still anything actionable here?

**Comment by a-robinson** [comment]
Yeah, there just hasn't been any push to do it since the issue was originally opened, so it's been prioritized behind other orchestration improvements.

**Comment by sonix07** [comment]
Since version 2.0.3 this is working for me. I  just added the  `--join=cockroachdb123:26257` to my swarm service where `cockroachdb123` needs to be the same as the docker swarm service name . I do something like `./cockroachdb init` inside the container and everything works. @a-robinson is this a good idea? I'm basically pointing the instance at itself, but I can scale the service and I see new nodes added to the Web-UI.
